src/query_loader/QueryLoader.cpp

```diff
@@ -17,47 +17,76 @@
 #include "Debug.h"
 #include "Equation.h"
 #include "File.h"
+#include "GlobalConfiguration.h"
 #include "InputQuery.h"
-#include "MStringf.h"
+#include "MarabouError.h"
 #include "MaxConstraint.h"
+#include "MStringf.h"
 #include "QueryLoader.h"
 #include "ReluConstraint.h"
-#include "GlobalConfiguration.h"
-#include "MarabouError.h"
-// TODO: add a unit test for a sanity check
 
 InputQuery QueryLoader::loadQuery( const String &fileName )
 {
     if ( !File::exists( fileName ) )
     {
-        throw MarabouError(MarabouError::FILE_DOES_NOT_EXIST, Stringf( "File %s not found.\n", fileName.ascii() ).ascii() );
+        throw MarabouError( MarabouError::FILE_DOES_NOT_EXIST, Stringf( "File %s not found.\n", fileName.ascii() ).ascii() );
     }
 
     InputQuery inputQuery;
     File input = File( fileName );
     input.open( File::MODE_READ );
 
     unsigned numVars = atoi( input.readLine().trim().ascii() );
-    unsigned numBounds = atoi( input.readLine().trim().ascii() );
+    unsigned numLowerBounds = atoi( input.readLine().trim().ascii() );
```

Comment:
make sure atoi is positive? 
I am not sure it's important, we are not writing defensive code and stuff might overflow, just popped to me (an easy change can be using a define that throes exception if it's not positive) ...

Reply:
I can add the define that would throw an exception, but it makes the code messier. There are also instances in the input parsers where the output of atoi is cast to unsigned without checking if the integer is negative. Do we want to add these protective checks when using atoi?

Reply:
Don't see why it's messy (will be just to call atoi_safe instead of atoi), but I agree, @guykatzz  what do you think?

Reply:
If you do something like `#define atoi_safe(a) a>=0?atoi(a):throw "Converting negative integer to unsigned"`, with the input being `input.readLine()`, the macro will do a double evaluation and call `input.readLine()` twice, since `a` appears twice in the macro. One way around this would be to save the output of `input.readLine()` to a variable before using the macro. Or maybe a better way would be to add functions to MString to do atoi and safe_atoi so the function could be defined once and called on any MString variable. This could be a separate issue / pull request?

Reply:
I don't feel very strongly either way. I think atoi() in general has been deprecated and should not be used, but as Kyle mentioned this isn't enforced in our code. I think in this case it would be enough to change the types to int and do an ASSERT that they are positive, but any solution you pick is fine.

Reply:
If we use ASSERT, does that only effect the code if DEBUG_ON is set? So if a user is running in release mode, the ASSERT macro has no effect and allows the program to continue even if negative ints are cast to unsigned ints?

Reply:
That's correct. If we wanted this to happen without DEBUG_ON, we would need to do an "if ... throw" instead of an assert.

Reply:
Ok makes sense. Do you have preference either way? Should we leave it as is and open an issue to address use of atoi to read unsigned ints throughout Marabou and continue discussion on the issue?

Reply:
Yes, that makes sense.